ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -674,9 +691,9 @@ - (NSDictionary *) powerState {
                    "You need to enable monitoring with `[UIDevice currentDevice].batteryMonitoringEnabled = TRUE`");
     }
 #endif
-#if RCT_DEV && TARGET_IPHONE_SIMULATOR && !TARGET_OS_TV
+#if RCT_DEV && TARGET_IPHONE_SIMULATOR && !TARGET_OS_TV && !TARGET_OS_VISION
     if ([UIDevice currentDevice].batteryState == UIDeviceBatteryStateUnknown) {
```

Comment:
Interesting, this API looks available on visionOS (Apple Vision Pro has battery)

Reply:
could be TARGET_OS_VISION_SIMULATOR then perhaps? I don't have the hardware myself so I can't say

Reply:
thanks for flagging this, the warning was correctly showing up in visionOs simulator.
`batteryLevel` would be "unknown" always in the simulator as per warning & the function should run fine on the actual device.

also, I think we can use `TARGET_OS_SIMULATOR` which should work for all simulators as per my understanding.

Reply:
if TARGET_OS_SIMULATOR is a superset of all the simulators to correctly have them avoid batteryState, that sounds great

Reply:
Just this one now @gupta-ji6 - are you testing this and/or do you want to switch to TARGET_OS_SIMULATOR?

Reply:
@mikehardy - I added `!TARGET_OS_TV` check to hide the warning on the simulator, but have removed it now since it was correct.

regarding `TARGET_IPHONE_SIMULATOR`, there are many other instances of it & it should work for us now. I'll replace it with `TARGET_OS_SIMULATOR` in the next PR & test it.

Reply:
Based on this comment - good to go when CI goes green:

> thanks for flagging this, the warning was correctly showing up in visionOs simulator.
batteryLevel would be "unknown" always in the simulator as per warning & the function should run fine on the actual device.

Reply:
> @mikehardy - I added `!TARGET_OS_TV` check to hide the warning on the simulator, but have removed it now since it was correct.
> 
> regarding `TARGET_IPHONE_SIMULATOR`, there are many other instances of it & it should work for us now. I'll replace it with `TARGET_OS_SIMULATOR` in the next PR & test it.

sounds great - I was comfortable with it after carefully re-reading your earlier comment. Any moment now this will be live via auto-release post merge, super cool! Thank you